### PR TITLE
Update openms to 2.6.0

### PR DESCRIPTION
**Pull request by jpfeuffer**
Update the openms package to 2.6

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/openms-meta/meta.yaml [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
+  url: https://abibuilder.informatik.uni-tuebingen.de/archive/openms/OpenMSInstaller/release/{{ version }}/OpenMS-{{ version }}-src.tar.gz
   sha256: af62010174f9e04d158fadfd4efdc5c3c9acc2b85a04ab2190ddd97a2f1ad7df
   patches:
     - apple_pyoms_nofixup.patch
     - init.py.patch
     - MANIFEST.in.patch
 
 build:
-  number: 6
+  number: 1
```
Comment (bgruening) [inline recipes/openms-meta/meta.yaml:16]: this should be 0

File: recipes/openms-meta/meta.yaml [hunk 2]
```diff
@@ -141,13 +141,10 @@ outputs:
         - {{ pin_compatible('sqlite', max_pin='x.x') }}
     test:
       commands:
-        - OPENMS_DATA_PATH=${PREFIX}/share/OpenMS/ ${PREFIX}/bin/FileMerger --help
+        - OPENMS_DATA_PATH=${PREFIX}/share/OpenMS/ ${PREFIX}/bin/OpenMSInfo
```
Comment (bgruening) [inline recipes/openms-meta/meta.yaml:144]: why is this change needed? Can you add both?

File: recipes/openms-meta/meta.yaml (hunk 1 above)
Comment (jpfeuffer) [reply recipes/openms-meta/meta.yaml:16]: Ah! Oops.

File: recipes/openms-meta/meta.yaml (hunk 2 above)
Comment (jpfeuffer) [reply recipes/openms-meta/meta.yaml:144]: Yes sure. Can be both. OpenMSInfo just gives some info about the build, too, to see if the settings were applied correctly.

Comment (bernt-matthias) [reply recipes/openms-meta/meta.yaml:144]: If the version information is in the output of `OpenMSInfo` you could try to match it with the package version.

Comment (bgruening) [reply recipes/openms-meta/meta.yaml:144]: I guess both is a good idea.

**Comment by bgruening** [comment]
@BiocondaBot please fetch artifacts

**Comment by bgruening** [comment]
The packages can be tested now. @jpfeuffer before we merge, can you please add those tests from above?

**Comment by bernt-matthias** [comment]
> The packages can be tested now

I can test them with the 2.6 Galaxy tools which are already in preparation https://github.com/galaxyproteomics/tools-galaxyp/pull/482

**Comment by bgruening** [comment]
@jpfeuffer why is the release not tagged? https://github.com/OpenMS/OpenMS/releases

**Comment by jpfeuffer** [comment]
We were waiting for the bioconda build in case there would be minor changes to be made ;)

**Comment by bgruening** [comment]
> We were waiting for the bioconda build in case there would be minor changes to be made ;)
> There is a draft release on github already (probably only seen by maintainers)

:heart: lovely. We take this as huge compliment :)

**Comment by bernt-matthias** [comment]
In the conda package there is again links (in the ctds that are produced by the binaries) that produces a 404

'404 Client Error: Not Found for url: https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Documentation/release/latest/html/UTILS_LowMemPeakPickerHiRes.html' accessing http://www.openms.de/documentation/UTILS_LowMemPeakPickerHiRes.html

404 Client Error: Not Found for url: https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Documentation/release/latest/html/UTILS_LowMemPeakPickerHiResRandomAccess.html' accessing http://www.openms.de/documentation/UTILS_LowMemPeakPickerHiResRandomAccess.html

Maybe it would be better to link not to latest but to the current version .. because its easier to keep those links stable.

**Comment by bernt-matthias** [comment]
For the Galaxy tests only 11 out of the 600+ tests fail. I guess I can check what's going wrong until Friday.

First thing I found is that the tests run MSGFPlusAdapter with 

```
Invalid parameter: -maxMissedCleavages.
```

Not sure if the error is on Galaxy / OpenMS side, but essentially we are executing the commands listed in the test CMake files.

**Comment by jpfeuffer** [comment]
> In the conda package there is again links (in the ctds that are produced by the binaries) that produces a 404
> 
> '404 Client Error: Not Found for url: https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Documentation/release/latest/html/UTILS_LowMemPeakPickerHiRes.html' accessing http://www.openms.de/documentation/UTILS_LowMemPeakPickerHiRes.html
> 
> 404 Client Error: Not Found for url: https://abibuilder.informatik.uni-tuebingen.de/archive/openms/Documentation/release/latest/html/UTILS_LowMemPeakPickerHiResRandomAccess.html' accessing http://www.openms.de/documentation/UTILS_LowMemPeakPickerHiResRandomAccess.html
> 
> Maybe it would be better to link not to latest but to the current version .. because its easier to keep those links stable.

Where do you see this? Where are the links coming from? In theory, yes there are also documentation for the old versions that we could link to.

> For the Galaxy tests only 11 out of the 600+ tests fail. I guess I can check what's going wrong until Friday.
> 
> First thing I found is that the tests run MSGFPlusAdapter with
> 
> ```
> Invalid parameter: -maxMissedCleavages.
> ```

This is from underlying MSGF, right? Are you using openms-thirdparty? If so, I need to bump the version there, let me check.

**Comment by bernt-matthias** [comment]
> Where do you see this?

This is in the ctd files generated from the binaries (which are the basis for the Galaxy tools). But the links should also be in the help text of the binaries (`-help` option).

I think the 2.6 version binaries should always link to the 2.6 docs and not latest. Because the set of tools changes and thereby we get 404 errors.

> Are you using openms-thirdparty?

Exactly

**Comment by jpfeuffer** [comment]
Would it be okay, if we leave the documentation until the next release? I assume it affects tools from the last version that were removed only? Maybe you could do a quick find/replace for the links in the CTDs?

**Comment by bernt-matthias** [comment]
> Maybe you could do a quick find/replace for the links in the CTDs?

Yes, I can do. Do you think it would be the best to replace latest by the version in general?

**Comment by jpfeuffer** [comment]
Yes I totally agree in general. See the linked PR 4984. I still have to implement something in case it is a version from develop to point to the nightly documentation.

**Comment by bgruening** [comment]
@BiocondaBot please fetch artifacts

**Review by timosachsenberg — APPROVED** [review]
tested and works

**Comment by timosachsenberg** [comment]
please merge

**Comment by timosachsenberg** [comment]
@BiocondaBot please add label

**Review by bgruening — APPROVED** [review]
Looks good to me. Go for it @jpfeuffer @timosachsenberg

**Comment by timosachsenberg** [comment]
@BiocondaBot please merge
